letsencrypt-apache/letsencrypt_apache/configurator.py

```diff
@@ -731,6 +732,67 @@ def enhance(self, domain, enhancement, options=None):
             logger.warn("Failed %s for %s", enhancement, domain)
             raise
 
+    def _set_http_header(self, ssl_vhost, header_name):
+        """Enables header header_name on ssl_vhost.
+
+        If header_name is not already set, a new Header directive is placed in
+        ssl_vhost's configuration with arguments from:
+        constants.HTTP_HEADER[header_name]
+
+        .. note:: This function saves the configuration
+
+        :param ssl_vhost: Destination of traffic, an ssl enabled vhost
+        :type ssl_vhost: :class:`~letsencrypt_apache.obj.VirtualHost`
+
+        :param header_name: a header name, e.g: Strict-Transport-Security
+        :type str
+
+        :returns: Success, general_vhost (HTTP vhost)
+        :rtype: (bool, :class:`~letsencrypt_apache.obj.VirtualHost`)
+
+        :raises .errors.PluginError: If no viable HTTP host can be created or
+            set with header header_name.
+
+        """
+        if "headers_module" not in self.parser.modules:
+            self.enable_mod("headers")
+
+        # Check if selected header is already set
+        self._verify_no_http_header(ssl_vhost, header_name)
+
+        # Add directives to server
+        self.parser.add_dir(ssl_vhost.path, "Header",
+                constants.HEADER_ARGS[header_name])
+
+        self.save_notes += ("Adding %s header to ssl vhost in %s\n" %
+                (header_name, ssl_vhost.filep))
+
+        self.save()
+        logger.info("Adding %s header to ssl vhost in %s", header_name,
+                ssl_vhost.filep)
+
+    def _verify_no_http_header(self, ssl_vhost, header_name):
+        """Checks to see if existing header_name header is in place.
```

Comment:
I agree. Changed `header_name` to `header_substring` and  `verify_no_http_header` to `verify_no_matching_http_header`.